cmd/catalyst-node/cluster/cluster.go

```diff
@@ -0,0 +1,239 @@
+package cluster
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	serfclient "github.com/hashicorp/serf/client"
+	"github.com/hashicorp/serf/cmd/serf/command/agent"
+	glog "github.com/magicsong/color-glog"
+	"github.com/mitchellh/cli"
+	"github.com/mitchellh/mapstructure"
+)
+
+type ClusterIface interface {
```

Comment:
The same comment as in the case of `Balancer` => I think you can call it `ClusterManager` and the implementation `ClusterManagerImpl`.